src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -255,6 +256,54 @@ public Icon getSystemIcon(File f) {
         }
     }
 
+   /**
+    * Icon for a file, directory, or folder as it would be displayed in
+    * a system file browser for the requested size.
+    * <p>
+    * The default implementation gets information from the
+    * <code>ShellFolder</code> class. Whenever possible, the icon
+    * returned will be a multi-resolution icon image,
+    * which will allow better scaling with different
+    * magnification factors.
+    * <p>
+    * Example: <pre>
+    * FileSystemView fsv = FileSystemView.getFileSystemView();
+    * Icon icon = fsv.getSystemIcon("application.exe", 64);
+    * JLabel label = new JLabel(icon);
+    * </pre>
+    *
+    * @param f a <code>File</code> object
+    * @param size width and height of the icon in pixels
```

Comment:
Pixels could be ambiguous now. Usually, Swing deals with user's space. That is 16×16 icon should actually be 32×32 if the scale factor of the current display is 200%.

Yes, this issue is somewhat irrelevant because the method returns multi-resolution icon. However, the terminology used must be unambiguous and clear; for consistency with other Swing API, it should be in terms of user's space coordinates, *virtual pixels*.

Reply:
Ok, fixed.